bitarray/sparse_bitarray.go

```diff
@@ -296,6 +300,67 @@ func (sba *sparseBitArray) IntersectsBetween(other BitArray, start, stop uint64)
 	return true
 }
 
+// WriteSparse serializes the sparseBitArray and passes it to the writer.
+func WriteSparse(w io.Writer, ba *sparseBitArray) error {
```

Comment:
Seems to me like the `BitArray` interface should just have `Serialize` and `Deserialize` methods exposed.